crates/web5/src/credentials/verifiable_credential_1_1.rs

```diff
@@ -128,85 +118,100 @@ pub struct VerifiableCredential {
-    credential_subject: Option<CredentialSubject>,
+#[derive(Default)]
+pub struct VerifiableCredentialCreateOptions {
+    pub id: Option<String>,
+    pub context: Option<Vec<String>>,
+    pub r#type: Option<Vec<String>>,
+    pub issuance_date: Option<SystemTime>,
+    pub expiration_date: Option<SystemTime>,
 }
 
 impl VerifiableCredential {
-    pub fn new(
-        id: String,
-        context: Vec<String>,
-        r#type: Vec<String>,
+    pub fn create(
         issuer: Issuer,
-        issuance_date: SystemTime,
-        expiration_date: Option<SystemTime>,
         credential_subject: CredentialSubject,
-    ) -> Self {
-        let context_with_base = std::iter::once(BASE_CONTEXT.to_string())
-            .chain(context.into_iter().filter(|c| c != BASE_CONTEXT))
-            .collect::<Vec<_>>();
+        options: Option<VerifiableCredentialCreateOptions>,
+    ) -> Result<Self> {
+        if issuer.to_string().is_empty() {
+            return Err(Web5Error::Parameter(String::from(
+                "issuer id must not be empty",
+            )));
+        }
 
-        let type_with_base = std::iter::once(BASE_TYPE.to_string())
-            .chain(r#type.into_iter().filter(|t| t != BASE_TYPE))
-            .collect::<Vec<_>>();
+        if let Issuer::Object(ref named_issuer) = issuer {
+            if named_issuer.name.is_empty() {
+                return Err(Web5Error::Parameter(String::from(
+                    "named issuer name must not be empty",
+                )));
+            }
+        }
+
+        if credential_subject.to_string().is_empty() {
+            return Err(Web5Error::Parameter(String::from(
+                "subject id must not be empty",
+            )));
+        }
+
+        let options = options.unwrap_or_default();
+
+        let context = {
+            let mut contexts = options
+                .context
+                .unwrap_or_else(|| vec![BASE_CONTEXT.to_string()]);
+
+            if !contexts.contains(&BASE_CONTEXT.to_string()) {
+                contexts.insert(0, BASE_CONTEXT.to_string());
+            }
+
+            contexts
+        };
+
+        let r#type = {
+            let mut types = options
+                .r#type
+                .unwrap_or_else(|| vec![BASE_TYPE.to_string()]);
+
+            if !types.contains(&BASE_TYPE.to_string()) {
+                types.insert(0, BASE_TYPE.to_string());
+            }
 
-        Self {
-            context: context_with_base,
+            types
+        };
+
+        let id = options
+            .id
+            .unwrap_or_else(|| format!("urn:uuid:{}", Uuid::new_v4()));
+
+        Ok(Self {
+            context,
             id,
-            r#type: type_with_base,
+            r#type,
             issuer,
-            issuance_date,
-            expiration_date,
+            issuance_date: options.issuance_date.unwrap_or_else(SystemTime::now),
+            expiration_date: options.expiration_date,
```

Comment:
should we let them create an "invalid" vc that is expired out of the gate.

I think what we have is fine actually, people using this should be aware of what they are doing, 

and once they sign and verify then it will instantly be invalid so they can catch it there but yea things to think about

Reply:
yeah I thought about that but I lean towards enabling them to create an date which is already expired, may be useful is weird edge cases, and the spec doesn't specifically make an assertion except, to your point, vc-jwt verification